### PR TITLE
Added Flake column

### PR DESCRIPTION
**Pull request by Meleeman01**
also added a request, a route, and an edit function in the slotsController. Flake column is not yet editable. model was edited to reflect changes in the Slot table.

### CODE REVIEW

File: laravel/app/Http/Requests/SlotEditRequest.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            
+            'flaked' => 'required',
```
Comment (itsrachelfish) [inline laravel/app/Http/Requests/SlotEditRequest.php]: These validation rules are not properly defined, there should be a single validation rule for `status` which uses a `in:...` rule.

File: laravel/app/Http/routes.php [hunk 2]
```diff
@@ -81,12 +81,15 @@
...
 Route::post('/schedule/{schedule}/delete', 'ScheduleController@delete');
 
-
-// Slot routes
-Route::get('/slot/{slot}/view', 'SlotController@view');
-Route::post('/slot/{slot}/take', 'SlotController@take');
-Route::post('/slot/{slot}/release', 'SlotController@release');
-
+//Routes for Admin
+Route::group(['middleware' => ['auth','admin']], function()
```
Comment (itsrachelfish) [inline laravel/app/Http/routes.php]: The admin middleware should only be used around the slot edit route. Adding it to all of these slot routes prevents normal users from signing up for shifts.

File: laravel/resources/js/admin.js [hunk 3]
```diff
@@ -14,6 +14,7 @@ $(document).ready(function()
     $('.upload-status').each(function()
     {
         $(this).value($(this).data('status'));
+        console.log($(this).value());
```
Comment (itsrachelfish) [inline laravel/resources/js/admin.js]: `console.log()` should be removed before this is merged

File: laravel/resources/js/admin.js [hunk 4]
```diff
@@ -65,15 +66,33 @@ $(document).ready(function()
...
 
         ajaxOptions.body = JSON.stringify(data);
-        fetch('/upload/' + upload + '/edit', ajaxOptions);
+        
+        console.log(upload,status);
+
+        // if there is nothing to upload, then check for volunteers who flaked
+        if (!upload) {
+            console.log('slot number: '+ slot);
+            console.log(fetch('/slot/'+ slot +'/edit',ajaxOptions));
```
Comment (itsrachelfish) [inline laravel/resources/js/admin.js]: Editing a slot has nothing to do with file uploads. The code to edit flake statuses should be completely separate from file uploads.

File: laravel/app/Http/Controllers/SlotController.php [hunk 5]
```diff
@@ -158,4 +158,11 @@ public function release(Request $request, Slot $slot)
 
         return redirect('/event/' . $slot->event->id);
     }
+
+    // change the flake column status
+    public function edit(Request $request, Slot $slot){
```
[the remark this answers is not in the document]
Comment (Meleeman01) [reply laravel/app/Http/Controllers/SlotController.php]: I think I tried it here, using SlotEditRequest, but if you think it should be here then I will put it there. one thing that was mentioned to me is that i should use telescope to see whats going on with the request.

**Comment by itsrachelfish** [comment]
@Meleeman01 I had to do some cleanup and minor tweaks before this could be merged:

- Removed console.log()
- Fixed indenting for status dropdown
- Updated status field to have more user friendly name / values
- Added default "Select One" option when no status is set

I also updated the javascript to automatically populate the saved value when you view a slot which was already edited.

**Comment by Meleeman01** [comment]
there was another console.log? yeah I was kinda at a loss for what to call the status fields

**Pull request merged**